knora/dsplib/utils/id_to_iri.py

```diff
@@ -0,0 +1,78 @@
+"""
+This module handles the replacement of internal IDs with their corresponding IRIs from DSP.
+"""
+import json
+import os
+from datetime import datetime
+from pathlib import Path
+
+from lxml import etree
+
+
+def id_to_iri(xml_file: str, json_file: str, out_file: str, verbose: bool) -> None:
+    """
+    This function replaces all occurrences of internal IDs with their respective IRIs inside an XML file. It gets the
+    mapping from the JSON file provided as parameter for this function.
+
+    Args:
+        xml_file : the XML file with the data to be replaced
+        json_file : the JSON file with the mapping (dict) of internal IDs to IRIs
+        out_file: path to the output XML file with replaced IDs (optional), default: "id2iri_replaced_" + timestamp + ".xml"
+        verbose: verbose feedback if set to True
+
+    Returns:
+        None
+    """
+
+    # check that provided files exist
+    if not os.path.isfile(xml_file):
+        print(f"File {xml_file} could not be found.")
+        exit(1)
+
+    if not os.path.isfile(json_file):
+        print(f"File {json_file} could not be found.")
+        exit(1)
+
+    # load JSON from provided json file to dict
+    mapping = json.load(open(json_file))
```

Comment:
`open()` should allways be called from a `with` block, to ensure closing the file, etc. Also I'd specify the mode and the encoding explicitely, otherwise it might break on windows.